src/qibo/backends/tensorflow.py

```diff
@@ -306,6 +306,20 @@ def entanglement_entropy(self, rho):
         entropy = self.np.sum(masked_eigvals * spectrum) / self.np.log(2.0)
         return entropy, spectrum
 
+    def calculate_norm(self, state, order=None):
+        state = self.cast(state)
```

Comment:
Maybe here you can directly use [`tf.norm`](https://www.tensorflow.org/api_docs/python/tf/norm), which seems to be quite similar to `np.linalg.norm`.

Reply:
I tried that too and implemented it but then moved back to `numpy` because `tf.norm` not only uses orders with different names, but also doesn't implement all the orders that `numpy` has. So, I kept using the `numpy` norm in order to unify the `order` parameter with all the other backends.

Reply:
> I tried that too and implemented it but then moved back to `numpy` because `tf.norm` not only uses orders with different names, but also doesn't implement all the orders that `numpy` has. So, I kept using the `numpy` norm in order to unify the `order` parameter with all the other backends.

Okay, I understand. The only problem I see is that this is not compatible with TensorFlow gradient descent in simulation. I Think it will lose track of the state using an unknown  (by TensorFlow) numpy function.

Reply:
> > I tried that too and implemented it but then moved back to `numpy` because `tf.norm` not only uses orders with different names, but also doesn't implement all the orders that `numpy` has. So, I kept using the `numpy` norm in order to unify the `order` parameter with all the other backends.
> 
> Okay, I understand. The only problem I see is that this is not compatible with TensorFlow gradient descent in simulation. I Think it will lose track of the state using an unknown (by TensorFlow) numpy function.

you mean if `backend.calculate_norm` is used as a callback?

Reply:
Yes, in general if you want to calculate gradients of a process which involves non-tensorflow objects, it can lose the track of the flow and return `None`.

Reply:
Do we have any test covering that right now? If so, shouldn't it be broken? If not, we need to have one then.

Reply:
I don't think we have tests for every scenario right now (and I don't think it is possible).

My only concern is that if we built the TensorFlow backend to do automatic differentiation, it makes sense to keep the total compatibility with TF, because I suppose if one decide to use `tensorflow` as backend is because some gradient descent has to be computed. 

With this in mind, maybe I would like to customize the method if `tensorflow` is set, in order to remove the `"order"` option and manually compute only one norm if TF is used.

Maybe @stavros11 has clearer picture of this.

Reply:
We could also just use `tf.norm` and let the user know that not all orders in the other backends are compatible with the `tensorflow` backend. It is non-uniform but it works.

Reply:
> We could also just use `tf.norm` and let the user know that not all orders in the other backends are compatible with the `tensorflow` backend. It is non-uniform but it works.

Yes, it is also a good solution.

Reply:
> Yes, it is also a good solution.

Done